queries/d/highlights.scm

```diff
@@ -1,275 +1,373 @@
-  "module" @keyword.import)
+; Members
+(aggregate_body
+  (variable_declaration
+    (declarator
+      (identifier) @variable.member)))
 
-(import_declaration
-  "import" @keyword.import)
+(property_expression
+  "."
+  (identifier) @variable.member)
 
-(type) @type
+(type
+  .
+  (_)
+  .
+  (identifier) @variable.member
+  .
+  (_))
 
-(catch_parameter
-  (qualified_identifier) @type)
+; Types
+(class_declaration
+  (class)
+  .
+  (identifier) @type)
 
-(var_declarations
-  (qualified_identifier) @type)
+(struct_declaration
+  (struct)
+  .
+  (identifier) @type)
 
-(func_declaration
-  (qualified_identifier) @type)
+(union_declaration
+  (union)
+  .
+  (identifier) @type)
 
-(parameter
-  (qualified_identifier) @type)
+(enum_declaration
+  (enum)
+  .
+  (identifier) @type)
 
-(class_declaration
+(alias_declaration
+  (alias)
+  .
   (identifier) @type)
 
-(fundamental_type) @type.builtin
+((identifier) @type
+  (#lua-match? @type "^[A-Z].*"))
 
-(module_fully_qualified_name
-  (packages
-    (package_name) @module))
+(type
+  .
+  (identifier) @type .)
 
-(module_name) @module
+[
+  (auto)
+  (void)
+  (bool)
+  (byte)
+  (ubyte)
+  (char)
+  (short)
+  (ushort)
+  (wchar)
+  (dchar)
+  (int)
+  (uint)
+  (long)
+  (ulong)
+  (real)
+  (double)
+  (float)
+  (cent)
+  (ucent)
+  (ireal)
+  (idouble)
+  (ifloat)
+  (creal)
+  (double)
+  (cfloat)
+] @type.builtin
 
-(at_attribute) @attribute
+; Functions
+(function_declaration
+  (identifier) @function)
 
-(user_defined_attribute
-  "@" @attribute)
+(call_expression
+  (identifier) @function)
 
-; Variables
-(primary_expression
-  "this" @variable.builtin)
+(call_expression
+  (type
+    (identifier) @function .))
+
+; Parameters
+(parameter
+  (_)
+  (identifier) @variable.parameter)
+
+(function_literal
+  "("
+  (type
+    (identifier) @variable.parameter))
+
+; Constructors
+(constructor
+  (this) @constructor)
+
+(destructor
+  (this) @constructor)
+
+(postblit
+  .
+  (this) @constructor)
+
+; Punctuation
+[
+  ";"
+  "."
+  ":"
+  ","
+  "=>"
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+"..." @punctuation.special
+
+; Ternaries
+(ternary_expression
+  [
+    "?"
+    ":"
+  ] @keyword.conditional.ternary)
+
+; Labels
+(label
+  (identifier) @label)
+
+(goto_statement
+  (identifier) @label)
+
+; Literals
+(string_literal) @string
+
+[
+  (int_literal)
+  (float_literal)
+] @number
+
+(char_literal) @character
+
+[
+  (true)
+  (false)
+] @boolean
+
+[
+  (null)
+  (special_keyword)
+] @constant.builtin
+
+; Comments
+(comment) @comment @spell
+
+((comment) @comment.documentation
+  (#lua-match? @comment.documentation "^///[^/]"))
+
+((comment) @comment.documentation
+  (#lua-match? @comment.documentation "^///$"))
+
+((comment) @comment.documentation
+  (#lua-match? @comment.documentation "^/[*][*][^*].*[*]/$"))
+
+((comment) @comment.documentation
+  (#lua-match? @comment.documentation "^/[+][+][^+].*[+]/$"))
```

Comment:
Wouldn't it be better to merge these?

Reply:
hmm, how so with lua match? `[+*]` could work but that'd allow for `/*+`

Reply:
Can you test the performance with multiple lua-match queries vs 1 or 2 match queries?